Applications/Spire/Source/BookView/BookViewModel.cpp

```diff
@@ -38,3 +59,41 @@ const std::shared_ptr<SecurityTechnicalsValueModel>&
     BookViewModel::get_technicals() const {
   return m_technicals;
 }
+
+void BookViewModel::on_operation(
+    const ListModel<BookQuote>::Operation& operation) {
+  visit(operation,
+    [&] (const auto& operation) {
+      m_preview_order_index.update(operation);
+    });
+}
+
+void BookViewModel::on_preview_order_update(
+    const optional<OrderFields>& order) {
+  auto get_quotes = [&] (Side side) {
```

Comment:
Write a model to implement this, in general you can't rely on arbitrary mutation of an existing model, it's best to treat polymorphic models as read only and use composition to augment them with new features.